### PR TITLE
fix: remove blue dot next to buy

### PR DESCRIPTION
**Pull request by LunrEclipse**
## Description

Removed blue dot next to buy for users that have completed the buy flow. 

_Linear ticket:_ WEB-2314
_Slack thread:_
_Relevant docs:_

### Before

Screenshot 2023-06-21 at 2 50 43 PM

### After

Screenshot 2023-06-21 at 2 50 57 PM

### CODE REVIEW

File: src/components/swap/SwapBuyFiatButton.tsx [hunk 1]
```diff
@@ -135,7 +128,6 @@ export default function SwapBuyFiatButton() {
       >
         <StyledTextButton onClick={handleBuyCrypto} disabled={buyCryptoButtonDisabled} data-testid="buy-fiat-button">
           <Trans>Buy</Trans>
-          {!buyFiatFlowCompleted && <Dot data-testid="buy-fiat-flow-incomplete-indicator" />}
```
Comment (vm) [inline src/components/swap/SwapBuyFiatButton.tsx:138]: are you able to remove the hook now? useBuyDiatFlowCompleted?

unclear if it's used elsewhere

File: src/components/swap/SwapBuyFiatButton.tsx [hunk 2]
```diff
@@ -46,7 +39,7 @@ const StyledTextButton = styled(ButtonText)`
 export default function SwapBuyFiatButton() {
   const { account } = useWeb3React()
   const openFiatOnRampModal = useOpenModal(ApplicationModal.FIAT_ONRAMP)
-  const [buyFiatFlowCompleted, setBuyFiatFlowCompleted] = useBuyFiatFlowCompleted()
+  const [, setBuyFiatFlowCompleted] = useBuyFiatFlowCompleted()
```
Comment (tinaszheng) [inline src/components/swap/SwapBuyFiatButton.tsx:42]: you can remove this too

**Review by tinaszheng — CHANGES_REQUESTED** [review]

File: src/components/swap/SwapBuyFiatButton.tsx (hunk 1 above)
Comment (LunrEclipse) [reply src/components/swap/SwapBuyFiatButton.tsx:138]: Believe so, just checked and the hook isn't used anywhere else!

File: src/state/user/hooks.tsx [hunk 3]
```diff
@@ -65,18 +64,6 @@ export function useUserLocaleManager(): [SupportedLocale | null, (newLocale: Sup
   return [locale, setLocale]
 }
 
-export function useBuyFiatFlowCompleted(): [boolean | undefined, (buyFiatFlowCompleted: boolean) => void] {
-  const dispatch = useAppDispatch()
-  const buyFiatFlowCompleted = useAppSelector((state) => state.user.buyFiatFlowCompleted)
-  const setBuyFiatFlowCompleted = useCallback(
-    (buyFiatFlowCompleted: boolean) => {
-      dispatch(updateUserBuyFiatFlowCompleted(buyFiatFlowCompleted))
```
Comment (vm) [inline src/state/user/hooks.tsx:73]: you can delete these actions now also

Comment (LunrEclipse) [reply src/state/user/hooks.tsx:73]: which actions?

**Review by tinaszheng — CHANGES_REQUESTED** [review]
so looks like you also want to remove the field `buyFiatFlowCompleted` from the user store, which is persisted through localStorage

make sure you also write a migration [here](https://github.com/Uniswap/interface/blob/main/src/state/user/reducer.ts#L172) to delete the property if it exists

**Review by vm — APPROVED** [review]

**Review by tinaszheng — APPROVED** [review]
🙌 
image

**Pull request merged**